contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -3839,17 +3840,19 @@ bbf_pivot(PG_FUNCTION_ARGS)
 	oldcontext = MemoryContextSwitchTo(tsql_outmost_context);
 	PG_TRY();
 	{
-		per_pivot_list = list_nth_node(List, tsql_outmost_estat->pivot_parsetree_list, tsql_outmost_estat->pivot_number - 1);
+		Assert(tsql_outmost_estat->pivot_parsetree_list && list_length(tsql_outmost_estat->pivot_parsetree_list) > 0);
+		per_pivot_list = list_nth_node(List, tsql_outmost_estat->pivot_parsetree_list, 0);
 		Assert(list_length(per_pivot_list) >= 2);
 		bbf_pivot_src_sql = list_nth_node(RawStmt, per_pivot_list, 0);
 		bbf_pivot_cat_sql = list_nth_node(RawStmt, per_pivot_list, 1);
+		query_string = list_nth(per_pivot_list, 2);
 	}
-	PG_FINALLY();
+	PG_CATCH();
 	{
-		MemoryContextSwitchTo(oldcontext);
+		PG_RE_THROW();
 	}
 	PG_END_TRY();
-
+	MemoryContextSwitchTo(oldcontext);
```

Comment:
It won't switch back to oldContext if it catch an excption ?

Reply:
Yes, we could add `MemoryContextSwitchTo(oldcontext);` at line 3852. I'm curious if we throw an error should we also take care of restoring to the oldcontext?

Reply:
In this case we should in this case
We're trying to follow Defensive programming.